kubespawner/spawner.py

```diff
@@ -1086,6 +1096,24 @@ def poll(self):
     def asynchronize(self, method, *args, **kwargs):
         return method(*args, **kwargs)
 
+    @async_generator
+    async def progress(self):
+        next_event = 0
+        self.log.debug('progress generator: %s', self.pod_name)
+
+        while True:
+            len_events = len(self.events.events)
+            if next_event < len_events:
+                for i in range(next_event, len_events):
+                    event = self.events.events[i]
+                    await yield_({
+                        'progress': 50,
+                        'message':  "%s [%s] %s" % (event.last_timestamp, event.type, event.message)
+                    })
+                next_event = len_events
+            await sleep(1)
```

Comment:
This while loop needs to end at some point. How do you signal that there will be no more events?